src/main/java/com/autotune/experimentManager/data/EMMapper.java

```diff
@@ -1,16 +1,23 @@
 package com.autotune.experimentManager.data;
 
+import com.autotune.common.experiments.ExperimentTrial;
+
+import java.util.HashMap;
 import java.util.LinkedList;
+import java.util.List;
 import java.util.concurrent.ConcurrentHashMap;
 
 public class EMMapper {
     private static volatile EMMapper emMapper = null;
     private volatile ConcurrentHashMap<String, ExperimentTrialData> emMap = null;
     private volatile ConcurrentHashMap<String, LinkedList<String>> emDeploymentRunIdMap = null;
 
+    private volatile ConcurrentHashMap<String, HashMap<String, ExperimentTrial>> emExpTrialMap = null;
+
     private EMMapper() {
         emMap = new ConcurrentHashMap<String, ExperimentTrialData>();
         emDeploymentRunIdMap = new ConcurrentHashMap<String, LinkedList<String>>();
+        emExpTrialMap = new ConcurrentHashMap<String, HashMap<String, ExperimentTrial>>();
```

Comment:
Change it to

    private volatile ConcurrentHashMap<String,  ExperimentTrial> emExpTrialMap = null;